index.js

```diff
@@ -1,59 +1,70 @@
-var fromFd = require('yauzl').fromFd;
-var collect = require('collect-stream');
-var bplistParse = require('bplist-parser').parseBuffer;
-var plistParse = require('plist').parse;
-var reg = require('./lib/reg');
-var once = require('once');
+const promisify = require('util').promisify;
+const fromFd = promisify(require('yauzl').fromFd);
+const collect = promisify(require('collect-stream'));
+const bplistParse = require('bplist-parser').parseBuffer;
+const plistParse = require('plist').parse;
+const reg = require('./lib/reg');
 
-var chrOpenChevron = 60;
-var chrLowercaseB = 98;
+const chrOpenChevron = 60;
+const chrLowercaseB = 98;
 
-module.exports = function(fd, cb){
-  var foundPlist = false;
-  cb = once(cb || function(){});
-
-  fromFd(fd, function(err, zip){
-    if (err) return cb(err);
-    var onentry;
-
-    zip.on('entry', onentry = function(entry){
-      if (!reg.test(entry.fileName)) {
-        return
-      } else {
-        foundPlist = true
-      }
-
-      zip.removeListener('entry', onentry);
-      zip.openReadStream(entry, function(err, file){
-        if (err) return cb(err);
+module.exports = async function (fd, { autoClose = true } = {}) {
+  const zip = await fromFd(fd, { autoClose });
+  zip.openReadStreamAsync = promisify(zip.openReadStream.bind(zip));
+  zip.on('error', (err) => {
+    throw err;
+  });
 
-        collect(file, function(err, src){
-          if (err) return cb(err);
+  const findInfo = async () => {
+    const { entry } = await findEntry(zip, reg.info.test.bind(reg.info));
+    if (!entry) {
+      return null;
+    }
+    const file = await zip.openReadStreamAsync(entry);
+    const src = await collect(file);
+    if (src[0] === chrOpenChevron) {
+      return plistParse(src.toString());
+    } else if (src[0] === chrLowercaseB) {
```

Comment:
im not a big fan of else after return, but dunno if the js coding style specifies anything about it